lib/rucio/core/rse_counter.py

```diff
@@ -138,6 +139,40 @@ def update_rse_counter(rse_id, session=None):
         update.delete(flush=False, session=session)
 
 
+@transactional_session
+def update_replica_counts(rse_id, session=None):
+    """
+    Read the replica counts and update rse_usage.
+
+    :param rse_id:   The rse_id to update.
+    :param session:  Database session in use.
+    """
+
+    # Get the counds of obsolete and deletably (expired) files
+    obsolete_files, obsolete_bytes = (session.query(func.count(), func.sum(models.RSEFileAssociation.bytes))
```

Comment:
I think @cserf mentioned it before, we need to pull in the hint here. Similar to 
https://github.com/rucio/rucio/blob/740b825cac747280ca159089bca4635264c07fc1/lib/rucio/core/replica.py#L1848
The issue is that the replicas table is not indexed. Although I am planing to add an rse_id index anyway, we have an index exactly for this kind of query (where tombstones are not null).